velox/functions/prestosql/Sequence.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include "velox/expression/DecodedArgs.h"
+#include "velox/expression/VectorFunction.h"
+#include "velox/vector/ConstantVector.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+// See documentation at https://prestodb.io/docs/current/functions/array.html
+class SequenceFunction : public exec::VectorFunction {
+ public:
+  static constexpr int32_t kMaxResultEntries = 10'000;
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    VectorPtr localResult;
+    try {
+      if (args[0]->type() == BIGINT()) {
+        localResult = applyNumber(rows, args, outputType, context);
+      } else {
+        VELOX_NYI(
+            "Unsupported type for argument of 'sequence' function: ",
+            args[0]->type()->toString());
+      }
+    } catch (const std::exception& e) {
```

Comment:
What types of errors are we catching here?

Reply:
I was surprised that we are catching a generic exception.
But the code uses current_exception and it seems that's what we do everywhere.
I believe this is to support "try".

Reply:
So without the `try catch` here, some tests are failing on github, and the errors are from ExpressionVerifier "Common eval wasn't supposed to throw, but it did. Aborting."

Then I compared my code with other functions and found they all have the `try catch`. After adding, the tests are passing